controllers/mirrorpeer_controller.go

```diff
@@ -127,7 +132,27 @@ func (r *MirrorPeerReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 	}
 
 	err = processMirrorPeerSecretChanges(ctx, r.Client, mirrorPeer)
-	return ctrl.Result{}, err
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+	tokensExchanged, err := r.checkTokenExchangeStatus(ctx, mirrorPeer)
+	if tokensExchanged {
+		logger.Info("Tokens exchanged", "MirrorPeer", mirrorPeer)
+		mirrorPeer.Status.Phase = multiclusterv1alpha1.ExchangedSecret
+		mirrorPeer.Status.Message = ""
+		statusErr := r.Client.Status().Update(ctx, &mirrorPeer)
+		return ctrl.Result{}, statusErr
+	}
+
+	if err != nil {
+		logger.Info("Tokens not exchanged", "MirrorPeer", mirrorPeer)
+		mirrorPeer.Status.Phase = multiclusterv1alpha1.ExchangingSecret
+		mirrorPeer.Status.Message = fmt.Sprintf("%v", err)
```

Comment:
```suggestion
		mirrorPeer.Status.Message = err.String()
```

should work

Reply:
This func is not there. I've changed it to `err.Error()`